File: extensions/shared-youtube/library/src/main/java/app/morphe/extension/shared/spoof/SpoofAppVersionPatch.java
/*
 * Copyright 2026 Morphe.
 * https://github.com/MorpheApp/morphe-patches
 *
 * Original hard forked code:
 * https://github.com/ReVanced/revanced-patches/commit/724e6d61b2ecd868c1a9a37d465a688e83a74799
 *
 * See the included NOTICE file for GPLv3 Section 7 terms that apply to Morphe contributions.
 */

package app.morphe.extension.shared.spoof;

import app.morphe.extension.shared.settings.SharedYouTubeSettings;

@SuppressWarnings("unused")
public class SpoofAppVersionPatch {

    public static String getDefaultTarget() {
        return "";
    }

    public static final boolean SPOOF_APP_VERSION_ENABLED = SharedYouTubeSettings.SPOOF_APP_VERSION.get();
    public static final String SPOOF_APP_VERSION_TARGET = SharedYouTubeSettings.SPOOF_APP_VERSION_TARGET.get();

    public static String getUniversalAppVersionOverride(String version) {
        return SPOOF_APP_VERSION_ENABLED
                ? SPOOF_APP_VERSION_TARGET
                : version;
    }

    public static boolean isSpoofingToLessThan(String version) {
        return SPOOF_APP_VERSION_ENABLED && SPOOF_APP_VERSION_TARGET.compareTo(version) < 0;
    }
}

File: patches/src/main/kotlin/app/morphe/patches/music/misc/spoof/appversion/SpoofAppVersionPatch.kt
/*
 * Copyright 2026 Morphe.
 * https://github.com/MorpheApp/morphe-patches/pull/1948
 *
 * See the included NOTICE file for GPLv3 Section 7 terms that apply to Morphe contributions.
 */

package app.morphe.patches.music.misc.spoof.appversion

import app.morphe.patches.music.misc.extension.sharedExtensionPatch
import app.morphe.patches.music.misc.settings.PreferenceScreen
import app.morphe.patches.music.misc.settings.settingsPatch
import app.morphe.patches.music.shared.Constants.COMPATIBILITY_YOUTUBE_MUSIC
import app.morphe.patches.shared.misc.settings.preference.ListPreference
import app.morphe.patches.shared.misc.spoof.appversion.baseSpoofAppVersionPatch

@Suppress("unused")
val spoofAppVersionPatch = baseSpoofAppVersionPatch(
    defaultTargetString = { "7.17.52" },
    preferenceScreen = PreferenceScreen.GENERAL,
    listPreference = {
        ListPreference(
            key = "morphe_spoof_app_version_target",
            entriesKey = "morphe_music_spoof_app_version_target_entries",
            entryValuesKey = "morphe_music_spoof_app_version_target_entry_values"
        )
    },
    block = {
        dependsOn(
            sharedExtensionPatch,
            settingsPatch
        )

        compatibleWith(COMPATIBILITY_YOUTUBE_MUSIC)
    }
)
